src/presence/presence-handler.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var C = require( '../constants/constants' ),
+	messageParser = require( '../message/message-parser' ),
+	messageBuilder = require( '../message/message-builder' );
+
+module.exports = class PresenceHandler {
+
+	constructor( options, connection, client ) {
+		this._options = options;
+		this._connection = connection;
+		this._client = client;
+		this._clientAddedCallback = null;
+		this._clientRemovedCallback = null;
+	}
+
+	getCurrentClients( callback ) {
+		this._presentClientsCallback = callback;
+		var msg = messageBuilder.getMsg( C.TOPIC.PRESENCE, C.ACTIONS.QUERY );
+		this._connection.send( msg );
+	}
+
+	onClientLogin( callback ) {
+		this._clientAddedCallback = callback;

```

Comment:
we would want to use an event emitter and do an `on` here and `off` somewhere. 

We should look at the API, since it might be better to do a 

`on( 'clientLoggedOn', () => {} )`

aware this is just first commit though!=D